nuxeo-features/nuxeo-platform-tag-service/nuxeo-platform-tag-core/src/test/java/org/nuxeo/ecm/platform/tag/AbstractTestTagService.java

```diff
@@ -194,12 +194,14 @@ public void testUntagOnTrash() {
         // trash doc
         Framework.getService(TrashService.class).trashDocument(file);
 
-        // wait for async tag removal
-        coreFeature.waitForAsyncCompletion();
+        // wait for async tag listener
+        txFeature.nextTransaction();
 
-        // check no more tag
+        // check tag is still present
+        file = session.getDocument(file.getRef());
```

Comment:
I see it's just a copy/paste of similar code above, but note that you can do just `file.refresh()` instead.

Reply:
👍 
Is this the right way to do it now? (just asking)
If yes, I will change my code and cleanup the whole file.

Reply:
They're equivalent but `refresh` is shorter and better shows intent so we might as well use it yes. Note that there's probably dozens of other test files with the old way too...

Reply:
👍 
Probably yes, but I will just handle this one 😬